global-script.lua

```diff
@@ -3088,4 +3148,87 @@ function tCompare(t1,t2)
     return table.concat(newTab,"|")
     end
     if cc2(t1) == cc2(t2) then return true else return false end
-end
\ No newline at end of file
+end
+
+
+function swapPlayerAreas(a, b)
+    local function tableSwap(table)
+        local temp = table[a]
+        table[a] = table[b]
+        table[b] = temp
+    end
+    local function tintSwap(table)
+        local oa = table[a]
+        local ob = table[b]
+        if type(oa) == "string" then
+            oa = getObjectFromGUID(oa)
+            ob = getObjectFromGUID(ob)
+        end
+        local ta = oa.getColorTint()
+        local tb = ob.getColorTint()
+        oa.setColorTint(tb)
+        ob.setColorTint(ta)
+        tableSwap(table)
+    end
+    local function positionSwap(table)
+        local oa = table[a]
+        local ob = table[b]
+        if type(oa) == "string" then
+            oa = getObjectFromGUID(oa)
+            ob = getObjectFromGUID(ob)
+        end
+        local ta = oa.getPosition()
+        local tb = ob.getPosition()
+        oa.setPosition(tb)
+        ob.setPosition(ta)
+    end
+
+    print("Player " .. a .. " is trading places with player " .. b .. ".")
+    for i = 1,2 do
```

Comment:
should this also not be in a local function like the other swaps?

Reply:
or maybe only `tableSwap` needs to be a local function?

Reply:
The locals here are more one-shot utility functions that exist simply because they're called twice.  (With the exception of tintSwap, which ends up only being called once.)  I'm not certain exactly what you're saying should be moved to be local.

If you're referring to the entire swapPlayerAreas(a, b) function: it COULD be in onSwapButtonClicked, but I wanted to keep it separate in case other things end up using it for whatever reason in the future.  That said, the notice that two players are swapping locations might be better suited to on the button itself -- because other uses may not want to send that message.

Reply:
so i guess it wasn't clear with the line this is started on, but this is specific to swapping the player hand transforms. should this not be a local function?

Reply:
I didn't really see the value in writing a local function for it.  I suppose it could be.  The others were mainly split out into function because the alternative involved copypasta (though tintSwap ends up only being called once in the final product).

Reply:
it's more coming from the standpoint of keeping this code in similar style for the feature